src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -646,6 +646,16 @@ static Linker nativeLinker() {
         /**
          * {@return a linker option used to denote the index of the first variadic argument layout in the
          *          function descriptor associated with a downcall linkage request}
+         * <p>
+         * The {@code index} value must be greater than zero, and less than or equal to the number of argument layouts
+         * of the function descriptor that is used in the same linkage request as this option. When the {@code index} is
+         * equal to the number of argument layouts in the descriptor, it indicates a variadic function to which zero
```

Comment:
This sentence is not clear. If index == num layouts then it means *all* arguments are variadic?

Reply:
index == 0 would mean all arguments are variadic. index == num layouts means none. (just like `"abcd".substring(4)` returns an empty string).

Reply:
Yes, that's what I meant and got confused. But I missed the all-important `zero` in the javadoc - so I think that this bit of text is ok as it is. Apologize for the confusion.